cliquet/storage/postgresql/__init__.py

```diff
@@ -586,9 +586,17 @@ def _format_conditions(self, filters, id_field, modified_field,
                 # Safely escape field name
                 field_holder = '%s_field_%s' % (prefix, i)
                 holders[field_holder] = filtr.field
-                # JSON operator ->> retrieves values as text.
-                # If field is missing, we default to ''.
-                sql_field = "coalesce(data->>:%s, '')" % field_holder
+                if filtr.operator in (COMPARISON.LT, COMPARISON.MAX,
+                                      COMPARISON.MIN, COMPARISON.GT):
+                    try:
+                        value = float(value)
+                        sql_field = "data->:%s" % field_holder
+                    except ValueError:
+                        sql_field = "coalesce(data->>:%s, '')" % field_holder
```

Comment:
It is here: https://github.com/mozilla-services/cliquet/pull/690/commits/6d6ca311abf7d64127504f1804f2943293a1fe63